src/Features/CSharp/Portable/UseExpressionBodyForLambda/UseExpressionBodyForLambdaHelpers.cs

```diff
@@ -168,21 +162,16 @@ private static LambdaExpressionSyntax WithExpressionBody(LambdaExpressionSyntax
             return updatedDecl;
         }
 
-        private static bool CreateReturnStatementForExpression(
-            SemanticModel semanticModel, LambdaExpressionSyntax declaration)
-        {
-            var lambda = (INamedTypeSymbol)semanticModel.GetTypeInfo(declaration).ConvertedType;
-            return !lambda.DelegateInvokeMethod.ReturnsVoid;
-        }
-
         private static LambdaExpressionSyntax WithBlockBody(
             SemanticModel semanticModel, LambdaExpressionSyntax declaration)
         {
             var expressionBody = GetExpressionBody(declaration);
+            var lambdaType = (INamedTypeSymbol)semanticModel.GetTypeInfo(declaration).ConvertedType;
```

Comment:
Wouldn't .GetSymbolInfo return the method here?

Reply:
There's a subtlety here, and comes down to things like error situations.  effectively, we need to know what the actual delegate type was, because we need to know if we should generate "return expr;" or just "expr;" inside the block we're creating.

We intentionally check and make sure the lambda converted to a real delegate type so we can make that determination.

If GetSymbolInfo doesn't actually validate that we properly converted to a delegate.  so we don't want to trust that.

Reply:
Does this code path execute without the other check though in my other comment ()?

Reply:
If not, then `ConvertedType` could be null and you'd get a null ref exception here

Reply:
> Does this code path execute without the other check though in my other comment

No, it should not :)   The logic is:   ```CanOfferUseExpressionBody -> ConvertToExpressionBody``` and ```CanOfferUseBlockBody -> ConvertToBlockBody```.  These helpers are only called if applicable.

Reply:
So it should be ok to use `GetSymbolInfo` then?

Reply:
in this case, yes.  but i'd prefer to keep both codepaths consistent TBH.

Reply:
makes sense